### PR TITLE
fix - fixing RequestHandler ensure failed SQS messages are processed

### PR DESCRIPTION
**Pull request by klcodanr**
Please ensure your pull request adheres to the following guidelines:
- [x] make sure to link the related issues in this description
- [x] when merging / squashing, make sure the fixed issue references are visible in the commits, for easy compilation of release notes

## Related Issues

https://trello.com/c/51CioK2G/289-requesthandler-swallows-failed-sqs-messages

Thanks for contributing!

### CODE REVIEW

File: packages/request-handler/src/index.js [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+      throw record.messageId;
+    }
+  }
+
+  /**
+   * Adds an additional wrap function to the invocation
+   * @param {import("@adobe/helix-shared-wrap").WrapFunction} fn
+   */
+  withWrapFunction(fn) {
+    this.#wrapFn = fn;
```
Comment (jdelbick) [inline packages/request-handler/src/index.js:222]: nitpick: do we want any security check to make sure it's a function?

File: packages/request-handler/test/index.test.js [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+      const requestHandler = new RequestHandler().withHandler('throw', () => {
+        throw new Error('Surprise!');
+      });
+      const main = requestHandler.getMain();
+      const res = await main(MOCK_REQUEST, mockContext({ action: 'throw' }));
+      assert.strictEqual(res.status, 500);
+    });
+
+    it('can supply wrapper function', async () => {
+      const requestHandler = new RequestHandler().withWrapFunction((fn) => fn);
```
Comment (jdelbick) [inline packages/request-handler/test/index.test.js:80]: nice to have: test checks the `fn` was called

**Review by jdelbick — APPROVED** [review]
LGTM

File: packages/request-handler/src/index.js (hunk 1 above)
Comment (klcodanr) [reply packages/request-handler/src/index.js:222]: I mean, it's an internal feature, you should find out pretty quickly that it doesn't work if you pass in something that's not a function.

**Comment by alexkli** [comment]
@klcodanr A couple notes post merge. I didn't see this PR before - would be great to include me in reviews at least for generic stuff:

1. It seems this is specifically for the extractor, so I would give it a more specific name like `extractor-handler` instead of the generic `request-handler`
2. In `getMain()` it does all the wrapping, i.e. the return is `return main.with(helixStatus).with(logger.trace).with(logger);`. This adds unrelated wrapper funtions (unrelated to the extractor request handler functionality) for the client without a way to change/opt out. I would actually turn this around and have the client pull in the wrap as is normal for how the wrap libraries are used in Franklin:

```
import { extractor } from '@adobe/contentlake-shared-request-handler';

function run(req, ctx) {
   // adjust inner syntax/details as necessary 
}

export const main = wrap(run)
  .with(extractor)
  .with(logger.trace)
  .with(logger)
  .with(helixStatus);
```

3. do not use helix-universal-logger. use `context.log.info/debug/etc()`, like [here for example](https://github.com/adobe/content-lake-frontegg-service/blob/b34fbcea4417cd0517840f671cdcda601921ba10/src/hooks/invite.js#L24-L39).
Helix also doesn't use it anymore. helix-universal (the code that is wrapped around the functions by helix deploy to make them work in e.g. Lambda) uses the [plain](https://github.com/adobe/helix-universal/blob/main/src/utils.js#L111) console.log functions which AWS Lamba logs with the levels as INFO/DEBUG etc. and then the [coralogix feeder](https://github.com/adobe/helix-coralogix-feeder/blob/content-lake/src/coralogix.js) (with small changes for our variant) will pass that to coralogix, preserving the log values. And not sending anything debug level or below (so it is only available in cloud watch) - which is something we can control in the feeder as needed.

cc @jdelbick @mattvryan

**Comment by jdelbick** [comment]
@alexkli 

> It seems this is specifically for the extractor, so I would give it a more specific name like extractor-handler instead of the generic request-handler

It is used in ingestor and some of those actions too

I agree with 2&3. Especially 3, we should not be using the helix univeral logger. Infact, we may be creating a very simple version of our own logger for special prefixes added to each message (jobId, requestId, error status etc. See my sample/POC of it here: https://github.com/adobe/content-lake-ingestor-dispatcher-service/pull/12)

**Pull request merged**